### DISCUSSION TITLE
Private View - Potential improvements

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#4969] [Feature Requests]
Hi -
currently starting my work on offering Smart industry solutions for small and medium businesses... and I decided to use Directus as base for doing so.

Initially I am tweaking the user experience... and adding some functionality I need. While I fully understand that you want to keep the code clean, I still thought its a good idea to share this with you... 

Video of improvements/changes:
https://user-images.githubusercontent.com/47870436/114296633-a6e4d880-9aac-11eb-9704-b2a2c42cbf43.mp4

Options for implementing this in directus are:  as new addtl. private view... or as options within the existing private view.

Let me know whether any particular changes are of interest for you... and I will send you the PR for those pieces.

BTW - thanks a lot for the great work you are doing. Really impressive!

KR
Roman

**Comment by benhaynes** [comment]
Hey Roman — this is super cool to see, thanks for sharing! Below are a few thoughts on the main changes I'm seeing:

* **Collapsible Nav Bar** — I like this idea! We already do this on smaller responsive viewports, but the option to hide it on larger screens is a nice idea. I think I'd like to find a way to integrate the "pin open/closed" into the left module bar without moving the user menu though. Not sure how often this will get used... could be on right-click of nav-bar, but maybe that's too hidden.
* **User Menu** — Not sure about this one. I think I understand why you did it, but my main concern is the _left to right_ hierarchy of the App. User is a top-level item, and therefore IMO belongs on the left along with modules. The right info bar is only meant to be for page context. Although maybe you're thinking that avatar is in the header, not the right bar. Either way, not sure about this one.
* **Header Bar** — Is currently fixed already, so the main difference here is color and padding... which can be adjustable in CSS overrides, and our soon-to-be-added "themes".
* **Smaller Info Sidebar Heading Height** — I like this, although it only works when the top "slot" is part of the header, and different from the sidebar components.

I'll keep thinking about some of this, but I think the collapsible nav bar is the main thing I'll try in the designs. 👍

**Comment by ernst86913** [comment]
Hi -
Actually I am also not happy with the user avatar on top right side... will change that again back to bottom left

Regarding hiding sticky / collapse toggle - I like that idea? Can you do some design for that?

Meanwhile I will be tweaking / separating my code a bit and will submit it as PR during next days

KR
Roman

Von meinem iPhone gesendet

**Reply by ernst86913** [reply]
Hi Ben,
attached you can find the update... let me know if thats fine for you... if yes I will submit it as PR (only the left side menu changes - not the ones on the right side... still working on that)....

KR
Roman

https://user-images.githubusercontent.com/47870436/114890278-2a832a00-9e0b-11eb-851e-4a93cb4328f8.mp4

**Reply by benhaynes** [reply]
Let me think through the placement and design of that icon. @rijkvanzanten do you think this functionality is a good direction? I'm on the fence, but don't mind it if it's not intrusive.

**Reply by rijkvanzanten** [reply]
I get that it could be nice for some layouts, but I'm a little worried about the UX and the hidden trigger that is the hover on the module bar. It's also a little strange to me that a hover of an unrelated icon in the module bar shows the menu for the currently active one 🤔 

Not sure if the slightly more space on the layout is worth it (80/20?)

**Reply by ernst86913** [reply]
@ Hover - I started working on having hover over a module item shows also the module menu of this particular item... but with the current way the application is designed thats a bit more tricky... just for your info

Would this addtl. functionality make more sense for you? @rijkvanzanten @benhaynes 

Just let me know... otherwise I keep it in my "private" branch.....

**Reply by benhaynes** [reply]
Updating the nav on module hover is certainly interesting, but I'd have to give it a _drive_ to know how that UX actually feels. From a tech perspective, I think this will likely be too big of a change though, as the navigations are quite heavy... and I'm not sure how much we gain from implementing that.

For now, let's keep this thread just about the show/hide/pin feature...

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
